sdk/python/feast/types.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2022 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Union
+
+from feast.protos.feast.types.Value_pb2 import ValueType as ValueTypeProto
+
+PRIMITIVE_FEAST_TYPES_TO_VALUE_TYPES = {
+    "INVALID": "INVALID",
+    "STRING": "STRING",
+    "BYTES": "BYTES",
+    "BOOL": "BOOL",
+    "INT32": "INT32",
+    "INT64": "INT64",
+    "FLOAT32": "FLOAT",
+    "FLOAT64": "DOUBLE",
+    "UNIX_TIMESTAMP": "UNIX_TIMESTAMP",
+}
+
+
+class FeastType(ABC):
+    """
+    A FeastType represents a structured type that is recognized by Feast.
+    """
+
+    def __init__(self):
+        """Creates a FeastType object."""
+        pass
+
+    @abstractmethod
+    def to_int(self) -> int:
+        """
+        Converts a FeastType object to the appropriate int value corresponding to
+        the correct ValueTypeProto.Enum value.
+        """
+        raise NotImplementedError
+
+
+class PrimitiveFeastType(Enum):
+    """
+    A PrimitiveFeastType represents a primitive type in Feast.
+    """
+
+    INVALID = 0
+    BYTES = 1
+    STRING = 2
+    BOOL = 3
+    INT32 = 4
+    INT64 = 5
+    FLOAT32 = 6
+    FLOAT64 = 7
+    UNIX_TIMESTAMP = 8
```

Comment:
Does this numbering match the proto equivalents?

Reply:
switched to match